### ISSUE TITLE
cannot build a vesta which runs on 64bit Linux systems

### ISSUE THREAD

**Issue by TEITechnicalCouncil** [TEIC/TEI#705]
Version numbers:

Ubuntu: 9.04

$ svn update
At revision 6701.

Description:

Obviously this can only be reproduced on a 64bit system.

(Go to the top of the vesta source tree.)
$ ant build-linux
[bunch of ant messages, no error]

$ java -jar deploy/linux/vesta.jar 
Exception in thread "main" java.lang.UnsatisfiedLinkError: Cannot load 32-bit SWT libraries on 64-bit JVM
at org.eclipse.swt.internal.Library.loadLibrary(Unknown Source)
at org.eclipse.swt.internal.Library.loadLibrary(Unknown Source)
at org.eclipse.swt.internal.C.&lt;clinit&gt;(Unknown Source)
at org.eclipse.swt.internal.Converter.wcsToMbcs(Unknown Source)
at org.eclipse.swt.internal.Converter.wcsToMbcs(Unknown Source)
at org.eclipse.swt.widgets.Display.&lt;clinit&gt;(Unknown Source)
at org.tei.vesta.Vesta.main(Vesta.java:534)

The error message is self-explanatory.

Workaround:

IMPORTANT NOTE: This is a fly-by-the-seat-of-your-pants workaround.  If you use this workaround or a similar workaround, you are likely to be eaten by a grue or sprout tentacles.

(I'm assuming you start at the top of vesta's source tree.)
$ pushd lib/swt-3.4-gtk-linux-x86/
$ mv swt.jar swt.jar.old
$ ln -s ~/src/azureus/swt.jar .
$ popd
$ ant build-linux
$ java -jar deploy/linux/vesta.jar
[Vesta starts normally...]

Basically. the above replaces the swt.jar shipped with vesta with one I know works on 64bit Linux.  

A proper solution would involve getting the 64bit swt.jar which exactly corresponds to the 32bit version.

Original comment by: @lddubeau
